packages/backup/src/Helper_Script_Manager.php

```diff
@@ -0,0 +1,322 @@
+		'If you no longer have Jetpack connected to your site, you can delete them manually.',
+		'If you have questions or need assistance, please contact Jetpack Support at https://jetpack.com/support/',
+		'If you like to build amazing things with WordPress, you should visit automattic.com/jobs and apply to join the fun – mention this file when you apply!;',
+	);
+
+	const INDEX_FILE = '<?php // Silence is golden';
+
+	/**
+	 * Installs a Helper Script, and returns its filesystem path and access url.
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @param string $script_body Helper Script file contents.
+	 * @return array|WP_Error     Either an array containing the path and url of the helper script, or an error.
+	 */
+	public static function install_helper_script( $script_body ) {
+		// Check that the script body contains the correct header.
+		if ( strncmp( $script_body, self::HELPER_HEADER, strlen( self::HELPER_HEADER ) ) !== 0 ) {
+			return new \WP_Error( 'invalid_helper', 'Invalid Helper Script header' );
+		}
+
+		// Refuse to install a Helper Script that is too large.
+		if ( strlen( $script_body ) > self::MAX_FILESIZE ) {
+			return new \WP_Error( 'invalid_helper', 'Invalid Helper Script size' );
+		}
+
+		// Create a jetpack-temp directory for the Helper Script.
+		$relative_temp_dir = self::create_temp_directory();
+		if ( is_wp_error( $relative_temp_dir ) ) {
+			return $relative_temp_dir;
+		}
+
+		// Generate a random filename, avoid clashes.
+		$max_attempts = 5;
+		for ( $attempt = 0; $attempt < $max_attempts; $attempt++ ) {
+			$file_key           = wp_generate_password( 10, false );
+			$relative_file_path = trailingslashit( $relative_temp_dir ) . 'jp-helper-' . $file_key . '.php';
+			$absolute_file_path = trailingslashit( ABSPATH ) . $relative_file_path;
+
+			if ( ! file_exists( $absolute_file_path ) ) {
+				// Attempt to write helper script.
+				if ( ! self::put_contents( $absolute_file_path, $script_body ) ) {
+					if ( file_exists( $absolute_file_path ) ) {
+						unlink( $absolute_file_path );
+					}
+
+					continue;
+				}
+
+				// Always schedule a cleanup run shortly after EXPIRY_TIME.
+				wp_schedule_single_event( time() + self::EXPIRY_TIME + 60, 'jetpack_backup_cleanup_helper_scripts' );
+
+				// Success! Figure out the URL and return the path and URL.
+				return array(
+					'path' => $absolute_file_path,
+					'url'  => trailingslashit( get_site_url() ) . $relative_file_path,
+				);
+			}
+		}
+
+		return new \WP_Error( 'install_faied', 'Failed to install Helper Script' );
+	}
+
+	/**
+	 * Given a path, verify it looks like a helper script and then delete it if so.
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @param string $path Path to Helper Script to delete.
+	 * @return boolean     True if the file is deleted (or does not exist).
+	 */
+	public static function delete_helper_script( $path ) {
```

Comment:
I wonder how many of these methods really have to be static.

Reply:
They're all static, because there is no instance of a `Helper_Script_Manager` object. I followed a similar pattern to many of the classes in the Jetpack Sync package, which are logical lumps of static methods which are easy to autoload. 

Would it be preferable to have each of these as a separate function just in the `Automattic\Jetpack\Backup` namespace?

Reply:
I see. If you feel these should be static and you don't consider having an instance of this object, I think they're good already as they are 👍